scio-core/src/main/scala/com/spotify/scio/values/ApproxFilter.scala

```diff
@@ -0,0 +1,53 @@
+package com.spotify.scio.values
+import com.spotify.scio.annotations.experimental
+import com.spotify.scio.coders.Coder
+
+/**
+ * A read only container for collections which can be used to check approximate
+ * membership queries.
+ *
+ * Constructors for [[ApproxFilter]] are defined using [[ApproxFilterBuilder]]
+ */
+trait ApproxFilter[-T] extends (T => Boolean) {
+
+  override def apply(t: T): Boolean = mayBeContains(t)
+
+  /**
+   * Check if the filter may contain a given element.
+   */
+  def mayBeContains(t: T): Boolean
+
+  /**
+   * Serialize the Filter to an Array[Byte]
+   */
+  def serialize: Array[Byte]
```

Comment:
Should probably add a `def writeTo(out: OutputStream)` for the FS API and rename this to `toBytes`?